pkg/inspector/image-inspector.go

```diff
@@ -195,6 +145,73 @@ func (i *defaultImageInspector) Inspect() error {
 	return nil
 }
 
+// createAndExtractImage creates a docker container based on the option's image with containerName.
+// It will then insepct the container and image and then attempt to extract the image to
+// option's destination path.  If the destination path is empty it will write to a temp directory
+// and update the option's destination path with a /var/tmp directory.  /var/tmp is used to
+// try and ensure it is a non-in-memory tmpfs.
+func (i *defaultImageInspector) createAndExtractImage(client *docker.Client, containerName string) (*docker.Image, error) {
+	container, err := client.CreateContainer(docker.CreateContainerOptions{
+		Name: containerName,
+		Config: &docker.Config{
+			Image: i.opts.Image,
+			// For security purpose we don't define any entrypoint and command
+			Entrypoint: []string{""},
+			Cmd:        []string{""},
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Unable to create docker container: %v\n", err)
+	}
+
+	// delete the container when we are done extracting it
+	defer func() {
+		client.RemoveContainer(docker.RemoveContainerOptions{
+			ID: container.ID,
+		})
+	}()
+
+	containerMetadata, err := client.InspectContainer(container.ID)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to get docker container information: %v\n", err)
+	}
+
+	imageMetadata, err := client.InspectImage(containerMetadata.Image)
+	if err != nil {
+		return imageMetadata, fmt.Errorf("Unable to get docker image information: %v\n", err)
+	}
+
+	if len(i.opts.DstPath) > 0 {
+		err = os.Mkdir(i.opts.DstPath, 0755)
+		if err != nil {
+			if !os.IsExist(err) {
+				return imageMetadata, fmt.Errorf("Unable to create destination path: %v\n", err)
+			}
+		}
+	} else {
+		// forcing to use /var/tmp because often it's not an in-memory tmpfs
+		i.opts.DstPath, err = ioutil.TempDir("/var/tmp", "image-inspector-")
+		if err != nil {
+			return imageMetadata, fmt.Errorf("Unable to create temporary path: %v\n", err)
+		}
+	}
+
+	reader, writer := io.Pipe()
+	go handleTarStream(reader, i.opts.DstPath)
```

Comment:
I was wondering that too.  We could throw a channel in there like you suggest and wait on the channel or a timeout.